starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/MainActivity.java

```diff
@@ -1,15 +1,32 @@
 package ly.generalassemb.drewmahrt.shoppinglistdetailview;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.support.v7.widget.LinearLayoutManager;
-import android.support.v7.widget.RecyclerView;
-
-import java.util.List;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AppCompatActivity;
 
 import ly.generalassemb.drewmahrt.shoppinglistdetailview.setup.DBAssetHelper;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends AppCompatActivity implements GroceryListFragment.OnItemSelectedListener{
+    public static final String ITEM_NAME_KEY = "item_name";
+    public static final String ITEM_DESCRIPTION_KEY = "item_description";
+    public static final String ITEM_PRICE_KEY = "item_price";
+    public static final String ITEM_CATEGORY_KEY = "item_category";
+
+
+    @Override
+    public void onItemSelected(ShoppingItem itemSelected) {
+        Bundle bundle = new Bundle();
+        bundle.putString(ITEM_NAME_KEY,itemSelected.getName());
+        bundle.putString(ITEM_DESCRIPTION_KEY,itemSelected.getDescription());
+        bundle.putString(ITEM_PRICE_KEY,itemSelected.getPrice());
+        bundle.putString(ITEM_CATEGORY_KEY,itemSelected.getType());
```

Comment:
One of the requirements was "Gather the details for the selected item from the database (not passed through the intent)". Technically, you're not passing these values via intent, but it's almost the same thing. This works OK in this case, but what if the item had 100 fields of data? Then you'd want to pass just the item ID here, and make a separate database query from the detail fragment to get the one item you need.